zulipterminal/helper.py

```diff
@@ -49,6 +50,12 @@
 StreamAccessType = Literal["public", "private", "web-public"]
 
 
+class SearchStatus(Enum):
+    DEFAULT = 0
+    FILTERED = 1
+    EMPTY = 2
```

Comment:
The enum is an improvement overall 👍 It acknowledges that there *are* actually three distinct states for the search status in the code.

Re implementation in this commit structure:
- this commit tracks it, but doesn't do anything *with* the extra state (cf. we track empty so we can display something different)
- this commit could migrate to an enum, then extend the enum in another commit, when you need to track the other state?
- it's strange in general seeing new asserts for `FILTERED`, but fewer changes for `EMPTY`, and none for `DEFAULT`; doing the split as per the above bullet would make it clearer where new state tracking was added and extra asserts added, vs renaming.

For the enum element names:
- `DEFAULT` is OK, but in terms of a search-status, it's not as clear as it could be if that's search-on or search-off. `UNFILTERED` is maybe too close to `FILTERED`, but is closer to the meaning of it?
- `EMPTY` is rather more `FILTERED_EMPTY`? We could end up in a state where it's "UNFILTERED_EMPTY" (DEFAULT_EMPTY), but I don't think the code covers that case?

Re location, helper.py is OK, though I'd prefer we reduce elements in here since 'helper' is such a generic name :)